### PR TITLE
Glossary edit for material ids

### PR DESCRIPTION
**Pull request by singima**
Added a couple sentences discussing how deal.II treats children cells upon refinement if the mesh is done in a parallel computing setting.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]
Personally, I don't think we should encourage to set material id, boundary id, ... on the active level; it just leads to problems in parallel during repartitioning (as discussed here), during refinement/coarsening also in serial, and in geometric multigrid if the ids on the levels are inconsistent to the active one. But banning this would be to complicated...

**Comment by singima** [comment]
Sorry for not being active in this conversation.  This is something that @tjhei and I looked at a while ago.  We adjusted step-40 to add material ids during the cell refinements and deal.II didn't respond well.

Maybe not necessary to comment here but just for context :D

**Review by peterrum — APPROVED** [review]
> We adjusted step-40 to add material ids during the cell refinements and deal.II didn't respond well.

Everyone does this mistake once. That's why I am trying to teach everyone to assign boundary ids and material ids on level 0.

**Comment by tjhei** [comment]
> hat's why I am trying to teach everyone to assign boundary ids and material ids on level 0.

Yes, but you will also need to assign them on all cells not only locally owned before doing any refinement...

Maybe this is worth mentioning. @singima please add the following:
"It is however safe to assign material ids consistently to all coarse cells on all MPI ranks before executing any refinement."

> We should communicate updated material ids through p4est for refinement. The story is more complicated with coarsening.

While I agree that we could do better, your suggestion is not something easy to fix as you would need to a) provide a mechanism to communicate ids to ghosts after somebody changes local ids, b) transfer ids on coarsen/refinement. At least this is not something I can assign to @singima right now.

**Review by drwells — APPROVED** [review]

File: doc/doxygen/headers/glossary.h [hunk 1]
```diff
@@ -1304,7 +1304,12 @@
  * either explicitly specify the material id, or if they don't, then GridIn
  * simply sets them to zero. Because the material of a cell is intended
  * to pertain to a particular region of the domain, material ids are inherited
- * by child cells from their parent upon mesh refinement.
+ * by child cells from their parent upon mesh refinement. However, if material
+ * ids are being assigned in a parallel distributed computation, any
+ * refinement or coarsening step will disregard the id assignments when
+ * a cell is moved to another process. So one must be careful when a mesh
+ * has more than one material id and they plan on using multiple processors
```
Comment (bangerth) [inline doc/doxygen/headers/glossary.h:1311]: This is slightly funny grammar. I think this matches better with the first half of the sentence:
```suggestion
 * has more than one material id and one plans on using multiple processors
```

**Comment by singima** [comment]
Made the edit @tjhei

**Review by drwells — APPROVED** [review]

**Comment by drwells** [comment]
Its just a documentation change so I will merge. Thanks again!

**Pull request merged**